src/ipc/ipc-zephyr.c

```diff
@@ -79,6 +83,51 @@ static bool message_handler(const struct device *dev, void *arg, uint32_t data,
 	return false;
 }
 
+#ifdef CONFIG_PM_DEVICE
+/**
+ * @brief IPC device suspend handler callback function.
+ * Checks whether device power state should be actually changed.
+ *
+ * @param dev IPC device.
+ * @param arg IPC struct pointer.
+ */
+static int ipc_device_suspend_handler(const struct device *dev, void *arg)
+{
+	struct ipc *ipc = (struct ipc *)arg;
+
```

Comment:
Why not just call ipc_get()?

Reply:
Well, this IPC Zephyr API was supposed to be generic to be used not only by Intel.
Thus we left some freedom for data passing in handler args.
Same is done in the IPC message handler. So if we pass anyway the IPC struct as function `arg`, why calling `ipc_get()`?

Reply:
The `generic  use` argument is enough for me.

Just passing it around seems weird when we have access to it locally. It looked as if it was done that way so that the `arg` parameter had some use.